features/device_key/source/DeviceKey.cpp

```diff
@@ -245,22 +245,26 @@ int DeviceKey::get_derived_key(uint32_t *ikey_buff, size_t ikey_size, const unsi
     return DEVICEKEY_SUCCESS;
 }
 
-int DeviceKey::generate_root_of_trust()
+int DeviceKey::generate_root_of_trust(size_t key_size)
 {
     int ret = DEVICEKEY_GENERATE_RANDOM_ERROR;
     uint32_t key_buff[DEVICE_KEY_32BYTE / sizeof(uint32_t)];
-    size_t actual_size = DEVICE_KEY_32BYTE;
+    size_t actual_size;
```

Comment:
In line 258, actual_size is reference-bound into read_key_from_kvstore as an in-out parameter.
read_key_from_kvstore reads the value of the size-parameter before writing the returned data size into it.
When actual_size is not initialized, this might lead to UB. Or am I missing something here?

Reply:
You are right. This has to be initialized to the size of key_buff.